### PR TITLE
[LOOP-2352] When no carb data found, return cob = 0

### PR DESCRIPTION
**Pull request by nhamming**
https://tidepool.atlassian.net/browse/LOOP-2352

when no cob is found, return cob = 0

### CODE REVIEW

File: LoopKit/CarbKit/CarbStore.swift [hunk 1]
```diff
@@ -926,7 +926,8 @@ extension CarbStore {
                 completion(.failure(error))
             case .success(let values):
                 guard let value = values.closestPrior(to: date) else {
-                    completion(.failure(.noData))
+                    // If we have no cob values in the store, and did not encounter an error, return 0
+                    completion(.success(CarbValue(startDate: date, quantity: HKQuantity(unit: self.preferredUnit, doubleValue: 0))))
```
Comment (ps2) [inline LoopKit/CarbKit/CarbStore.swift]: Maybe `HKQuantity(unit: self.preferredUnit, doubleValue: 0)` -> `HKQuantity(unit: .milligramsPerDeciliter, doubleValue: 0)`; this shouldn't depend on the user's preferred unit, I think. Doesn't really cause any issues, but it's an unneeded dependency that might complicate refactoring?

**Review by ps2 — APPROVED** [review]

**Review by darinkrauss — APPROVED** [review]
LGTM! 🍔

File: LoopKit/CarbKit/CarbStore.swift (hunk 1 above)
Comment (nhamming) [reply LoopKit/CarbKit/CarbStore.swift]: @ps2 this is COB

Comment (ps2) [reply LoopKit/CarbKit/CarbStore.swift]: Sorry, I mean .grams, then. :)

Comment (nhamming) [reply LoopKit/CarbKit/CarbStore.swift]: will update all instances to `.gram()`

**Pull request merged**